debugutils/logs.go

```diff
@@ -0,0 +1,152 @@
+package debugutils
+
+import (
+	"fmt"
+	"io"
+	"path/filepath"
+
+	"github.com/solo-io/go-utils/installutils/helmchart"
+	"github.com/solo-io/go-utils/installutils/kuberesource"
+	"github.com/solo-io/go-utils/kubeutils"
+	"github.com/spf13/afero"
+	"golang.org/x/sync/errgroup"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	corev1client "k8s.io/client-go/kubernetes/typed/core/v1"
+	"k8s.io/client-go/rest"
+)
+
+type LogsRequest struct {
+	podMeta       metav1.ObjectMeta
+	containerName string
+	request       *rest.Request
+}
+
+type LogsResponse struct {
+	podMeta       metav1.ObjectMeta
+	containerName string
+	reader        io.ReadCloser
+}
+
+func (lr *LogsRequest) BuildId(dir string) string {
+	return filepath.Join(dir, fmt.Sprintf("%s_%s_%s.log", lr.podMeta.Namespace, lr.podMeta.Name, lr.containerName))
+}
+
+func NewLogsRequest(podMeta metav1.ObjectMeta, containerName string, request *rest.Request) *LogsRequest {
+	return &LogsRequest{podMeta: podMeta, containerName: containerName, request: request}
+}
+
+type LogStorageClient struct {
+	fs  afero.Fs
+	dir string
+}
+
+func NewLogFileStorage(fs afero.Fs, dir string) *LogStorageClient {
+	return &LogStorageClient{fs: fs, dir: dir}
+}
+
+func (lfs *LogStorageClient) FetchLogs(requests []*LogsRequest) error {
```

Comment:
nit: suspect `lfs` was a pre-refactor artifact, not sure why we'd use it as shorthand.